src/themes/from_zed_theme.rs

```diff
@@ -1,210 +1,285 @@
+const ZED_THEME_SUMMERCAMP_URL: &str = "https://raw.githubusercontent.com/zed-industries/zed/main/assets/themes/summercamp/summercamp.json";
+
 #[derive(serde::Deserialize)]
 struct ZedThemeManiftest {
     themes: Vec<ThemeContent>,
 }
 
-pub fn from_zed_theme(url: &str) -> anyhow::Result<Vec<Theme>> {
-    let json_str = cache_download(
-        url,
-        "zed-themes",
-        &std::path::PathBuf::from(url)
-            .file_name()
-            .unwrap_or_else(|| panic!("The url ({:?}) should contain file name.", url))
-            .to_string_lossy(),
-    )?;
-    let manifest: ZedThemeManiftest = serde_json5::from_str(&json_str).unwrap_or_else(|error| {
-        panic!("Cannot parse JSON downloaded from {url:?} due to:\n{error:#?}")
-    });
+/// Get all known Zed themes as a `ThemeDescriptor`.
+pub(crate) fn theme_descriptors() -> Vec<ThemeDescriptor> {
+    [
+        ThemeDescriptor::ZedThemeURLMap("Andromeda", ZED_THEME_ANDROMEDA_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Cave Dark", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Cave Light", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Dune Dark", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Dune Light", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Estuary Dark", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Estuary Light", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Forest Dark", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Forest Light", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Heath Dark", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Heath Light", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Lakeside Dark", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Lakeside Light", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Plateau Dark", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Plateau Light", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Savanna Dark", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Savanna Light", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Seaside Dark", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Seaside Light", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Sulphurpool Dark", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Atelier Sulphurpool Light", ZED_THEME_ATELIER_URL),
+        ThemeDescriptor::ZedThemeURLMap("Ayu Dark", ZED_THEME_AYU_URL),
+        ThemeDescriptor::ZedThemeURLMap("Ayu Light", ZED_THEME_AYU_URL),
+        ThemeDescriptor::ZedThemeURLMap("Ayu Mirage", ZED_THEME_AYU_URL),
```

Comment:
We can reduce the repetition of `ThemeDescriptor::ZedThemeURLMap` if we pack the arguments into tuples:
```rs
[
   // ...
   ("Ayu Dark", ZED_THEME_AYU_URL),
   ("Ayu Mirage", ZED_THEME_AYU_URL),
   // ...
]
    .into_iter()
    .map(ThemeDescriptor::ZedThemeURLMap) 
    // If point free style doesn't work due to borrowing, do this:
    // .map(|(name, url)| ThemeDescriptor::ZedThemeURLMap(name, url))
    .collect_vec()
```